### PR TITLE
Leverage fs-extra instead of defining our own fsUtils

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
Also:
1. Use secure random string
1. Slightly modify newline rules that are being annoying
1. Remove '--type-check' from lint command since it's deprecated and no longer necessary

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -207,15 +207,15 @@
     "scripts": {
         "vscode:prepublish": "tsc -p ./",
         "compile": "tsc -watch -p ./",
-        "lint": "tslint --project tsconfig.json -e src/*.d.ts --type-check -t verbose",
+        "lint": "tslint --project tsconfig.json -e src/*.d.ts -t verbose",
```
Comment (StephenWeatherford) [inline package.json]: why this change?

Comment (EricJizbaMSFT) [reply package.json]: I was getting this message when I ran the lint task:
```
--type-check is deprecated. You only need --project to enable rule which need type information.
```
